Python/optimizer.c

```diff
@@ -182,6 +182,12 @@ _PyOptimizer_Optimize(
     if (err <= 0) {
         return err;
     }
+
+    if (++interp->executors_created >= JIT_CLEANUP_THRESHOLD) {
+        interp->executors_created = 0;
```

Comment:
You could move this into `_Py_Executors_InvalidateCold`. That way, calling `_Py_Executors_InvalidateCold` does not require you do pre-update this value.

The reason I'm suggesting this is that `_Py_Executors_InvalidateAll` is already modifying `interp->executors_created`, so I think `_Py_Executors_InvalidateCold` should also be responsible for this.